doc/userguide/rules/vlan-keywords.rst

```diff
@@ -0,0 +1,92 @@
+VLAN Keywords
+=============
+
+.. role:: example-rule-action
+.. role:: example-rule-header
+.. role:: example-rule-options
+.. role:: example-rule-emphasis
+
+vlan.id
+-------
+
+Suricata has a ``vlan.id`` keyword that can be used in signatures to identify
+and filter network packets based on Virtual Local Area Network IDs. By default,
+it matches all layers if a packet contains multiple VLAN layers. However, if a
+specific layer is defined, it will only match that layer.
+
+Syntax::
+
+ vlan.id: [op]id[,layer];
+
+The id can be matched exactly, or compared using the ``op`` setting::
+
+ vlan.id:300    # exactly 300
+ vlan.id:<300,0   # smaller than 300 at layer 0
+ vlan.id:>=200,1  # greater or equal than 200 at layer 1
+
+vlan.id uses :ref:`unsigned 16-bit integer <rules-integer-keywords>`.
+
+.. table:: **Id values for vlan.id keyword**
+
+    ========  ================================================
+    Value     Description
+    ========  ================================================
+    1 - 4094  Valid range for vlan id
+    0 - 3     Valid range of number of layers (with ``count``)
+    ========  ================================================
+
+This keyword also supports ``all`` and ``count`` as arguments for ``layer``.
+``all`` matches only if all VLAN layers match and ``count`` matches based on
+the number of layers.
+
+.. table:: **Layer values for vlan.id keyword**
+
+    ===============  ================================================
+    Value            Description
+    ===============  ================================================
+    [default]        Match all layers
+    0 - 2            Match specific layer
+    ``-3`` - ``-1``  Match specific layer with back to front indexing
+    all              Match only if all layers match
+    count            Match on the number of layers
```

Comment:
while I like that we can match on this, I do question integrating it with a `vlan.id` keyword. Should we have a `vlan.layers` keyword instead? I feel that these options are not only not about the `id`, they also bring their own syntax.

Reply:
Ok.

About naming, should we have a unique naming scheme `count` for such keywords ?
Like `vlan.layers.count` `ldap.responses.count`...